tests/integrationv2/common.py

```diff
@@ -111,7 +111,16 @@ def __init__(self, name, prefix, location=TEST_CERT_DIRECTORY):
             self.algorithm = 'RSAPSS'
 
     def compatible_with_cipher(self, cipher):
-        return (self.algorithm == cipher.algorithm) or (cipher.algorithm == 'ANY')
+        if self.algorithm == cipher.algorithm:
+            return True
+        if cipher.algorithm == 'ANY':
+            return True
+        if self.algorithm == 'RSAPSS':
+            if cipher.algorithm != 'RSA':
```

Comment:
I _think_ the reason we have to do this is because `RSA` in a cipher is a indicator of RSA Key Exchange? But cipher naming is tricky, so maybe worth including a comment about this?

Reply:
Added comments

Reply:
But yes, it's also VERY unclear here that "algorithm" would represent the auth method instead of the key exchange or encryption algorithms -_-